src/compiler/program.ts

```diff
@@ -744,16 +744,20 @@ function formatCodeSpan(file: SourceFile, start: number, length: number, indent:
 }
 
 /** @internal */
-export function formatLocation(file: SourceFile, start: number, host: FormatDiagnosticsHost, color = formatColorAndReset) {
-    const { line: firstLine, character: firstLineChar } = getLineAndCharacterOfPosition(file, start); // TODO: GH#18217
+export function formatLocation(file: SourceFile, start: number, host: FormatDiagnosticsHost, color = formatColorAndReset, withPosition = true) {
     const relativeFileName = host ? convertToRelativePath(file.fileName, host.getCurrentDirectory(), fileName => host.getCanonicalFileName(fileName)) : file.fileName;
 
     let output = "";
     output += color(relativeFileName, ForegroundColorEscapeSequences.Cyan);
-    output += ":";
-    output += color(`${firstLine + 1}`, ForegroundColorEscapeSequences.Yellow);
-    output += ":";
-    output += color(`${firstLineChar + 1}`, ForegroundColorEscapeSequences.Yellow);
+
+    if (withPosition) {

```

Comment:
I noticed in #52328 that the symbol output elides these numbers already by replacing them with `--`; maybe that's a better idea than omitting it entirely?

Reply:
1. `--` is a better idea.
2. Adding a new overload is worth reflecting in the baselines IMO.